Cgm -> TaC

ClumsyAlien opened the pull request:
After creating our fork of Mr.Crayfish's Gun mod we now need to port all of our current changes and updates over, one small step at a time!

ClumsyAlien committed:
Updated starting naming conventions

ClumsyAlien committed:
Update mods.toml

ClumsyAlien committed:
Pretty much a test

ClumsyAlien committed:
cgm -> tac

Now working, carefully changed MODID

ClumsyAlien committed:
Added full - Per weapon adjustability of Horizontal recoil

ClumsyAlien committed:
Animation Files +^

Ported to Official mappings and cleaned

ClumsyAlien committed:
DO NOT PULL  THIS UPDATE

Major issue on my local end with loading world, much code and event Controllable have been either removed / commented out, world stuck at 100% loading and is unable to open / recreate a new world

Uncertain if this issue is due to any part of my code however, I am resetting my local project in order to perform a few more tests cleanly.

THIS BUILD IS MISING MANY FEATURES AND MAY EVEN BE NON FUNCTIONAL

ClumsyAlien committed:
index on ClumsyAlien-LocalProjectRunClient-Issue: 3a17f44 DO NOT PULL  THIS UPDATE

ClumsyAlien committed:
On ClumsyAlien-LocalProjectRunClient-Issue: !!GitHub_Desktop<origin/Porting-TaC>

ClumsyAlien committed:
Semi working select fire, currently works for Semi and Safe

ClumsyAlien committed:
Merge branch 'Porting-TaC' into commit_rescued2

ClumsyAlien committed:
Merge pull request #3 from ClumsyAlien/commit_rescued2

Implementing select fire baseline, at least semi correctly...

aemogie commented on src/main/java/com/tac/guns/entity/ProjectileEntity.java:
```diff
@@ -37,7 +36,6 @@
 import net.minecraft.entity.EntityType;
 import net.minecraft.entity.LivingEntity;
 import net.minecraft.entity.Pose;
-import net.minecraft.entity.item.ArmorStandEntity;
```
is this cz of IntelliJ's "auto-optimize imports"?

ClumsyAlien replied:
I would think, if this is projectile entity this must be from where I started to test more complex bullet physics (Will be creating a separate branch for shortly), this is still an auto import and will be cleared before final push.

aemogie commented on src/main/java/com/tac/guns/client/render/gun/model/ak47_animation.java:
```diff
@@ -0,0 +1,53 @@
+            float cooldownOg = tracker.getCooldownPercent(stack.getItem(), Minecraft.getInstance().getFrameTime());
+
+            // Math provided by Bomb787 on GitHub and Curseforge!!!
+            matrices.translate(0, 0, 0.190f * (-4.5 * Math.pow(cooldownOg-0.5, 2) + 1));
```
@Bomb787 can u explain what exactly is going here XD
I probably can reduce code duplication but don't want to mess with stuff until I understand what's going on.

ClumsyAlien replied:
Sooooooooooooo ```cooldownOg``` is really just the percentage through the original cooldown of an item, and all we do is just splice the percentage into the equation, ``` 0.190f ``` is our multiplier for the equation.

aemogie commented on src/main/java/com/tac/guns/client/render/gun/model/m1851_animation.java:
```diff
@@ -0,0 +1,79 @@
+        //Always pop
+        matrices.popPose();
+    }
+    private double easeInOutBack(double x) {
```
make this static maybe? it doesn't use any fields so there's no real reason for it to be non-static.

ClumsyAlien replied:
Good point, 
**Idea** possibly create a ```FireAnimationHelper``` class to handle these sort of methods. 
In short yes ```easeInOutBack``` can be made static.

ClumsyAlien committed:
Working Select Fire (Beta)

Thanks @Aemogie :heart: Literally one word was holding the rest of the code back!

Updated all the guns with basic fire modes, tested and working.

ClumsyAlien committed:
Added fire mode indication to Item Description for TimelessGunItem

ClumsyAlien committed:
Attempted compile client reflection errors, Added and confirmed working sounds

ClumsyAlien committed:
Alpha sight testing, cleaned weird stuff in folders

Much sight testing, all still non final as I attempt to figure out the best way to support our high-res scopes

286799714 committed:
Massive crosshair overhaul, Alpha animation testing, Horizontal recoil rework

* update Dynamic Crosshair

* Correctly transfer part of onClientTick() in CrosshairHandler

* Added scaling based on Inaccuracy and Recoil (On weapon firing)

* Make dynamic crosshair pan instead of scale

* ar_15_p is animated as an example.

* Updated Dynamic Crosshair with all modifiers, refined texture, and completely updated / fixed Horizontal recoil per weapon!

* some animation .json files changed

Co-authored-by: Clumsy Alien <therealclumsyalien@gmail.com>

ClumsyAlien committed:
Reworked recoil a bit, horizontal modifiers are accessible and used however not show under `Perks:`, first working prototype of Ricochet

ClumsyAlien committed:
Added new gun option for bullet ricochet

ClumsyAlien committed:
Now checks for material type, currently we statically support 4, will look into extendible versions

ClumsyAlien committed:
Added existing crafting recipes to the mod along with fixing table crash bug

ClumsyAlien committed:
New Guns, Inspection screen functional, new animations, refined weapons, refined the attachment screen a tad, Fixed the workbench for crafting weapons found inside of it, still an issue with adding colors via Vanilla Crafting Table

ClumsyAlien committed:
Many new weapons, updates to libraries and classes, much much more

ClumsyAlien committed:
Reverted all gradle versions and removed geckolib alpha tests

ClumsyAlien committed:
Injected Neko's weapon balancing, some model updates, some build edits, lots of new adjustments to dynamic crosshair

ClumsyAlien committed:
Attempts at Zoom[] and List<Zoom> (#8)

ClumsyAlien committed:
Weapon UI and overall changes (#9)

* Alpha stage weapon UI, positional tweaking required, scopes and more

* Now statically scales according to GUI size

* Scope attachment, basic breathing built by MaydayMemory!!!

* Added much of Mayday's impressive work!

* Bunch of UI updates, scope updates, and attachment inventory updates

* Config option support for rendering, safety and breathing

* All weapon updates along with revised GUI handling via Mayday!!!!!!!

Thank Mayday big time for the time and effort he has put in to help the mod!!!

* Complete rework of our GUI, config updates to support, reactions to barrel attachments.

* Added Muzzle Flash Thank you Mayday once again!!!!!!!!! Added LPVO scope completely along with HUD config settings

* Huge updates all round

* Complete for public version 0.1.0 !

ClumsyAlien committed:
Temporary sounds cleanup

ClumsyAlien commented:
An absolutely beautiful development period, 1.16.X is and will now stay as our Main home, 1.16.X public versions of the mod will be found here, all work must be done and only merged into Main, never work directly into Main.
